ios-drag-drop.js

```diff
@@ -136,8 +136,10 @@
       var touch = event.changedTouches[0];
       var x = touch[coordinateSystemForElementFromPoint + 'X'];
       var y = touch[coordinateSystemForElementFromPoint + 'Y'];
-      dropEvt.offsetX = x - target.x;
-      dropEvt.offsetY = y - target.y;
+      dropEvt.clientX = x;
+	  dropEvt.clientY = y;
+	  dropEvt.offsetX = x - event.target.clientLeft;
+	  dropEvt.offsetY = y - event.target.clientTop;
```

Comment:
I tested the switch to use target.clientXXX everything works out the same, so I have switched the code.

 

From: Kaur Kuut [mailto:notifications@github.com] 
Sent: Friday, January 23, 2015 4:24 PM
To: timruffles/ios-html5-drag-drop-shim
Cc: Jerel R Hass
Subject: Re: [ios-html5-drag-drop-shim] fixed image hover in left corner and client x-y (#31)

 

In ios-drag-drop.js <https://github.com/timruffles/ios-html5-drag-drop-shim/pull/31#discussion_r23481352> :

> @@ -136,8 +136,10 @@
>        var touch = event.changedTouches[0];
>        var x = touch[coordinateSystemForElementFromPoint + 'X'];
>        var y = touch[coordinateSystemForElementFromPoint + 'Y'];
> -      dropEvt.offsetX = x - target.x;
> -      dropEvt.offsetY = y - target.y;
> +      dropEvt.clientX = x;
> +       dropEvt.clientY = y;
> +       dropEvt.offsetX = x - event.target.clientLeft;
> +       dropEvt.offsetY = y - event.target.clientTop;

If event.target.clientLeft here can be replaced with target.clientLeft then I think we should do that.

I haven't looked too closely at how (if at all) the targets differ here, but I do see that we spend some considerable effort deriving target via elementFromTouchEvent.

—
Reply to this email directly or view it on GitHub <https://github.com/timruffles/ios-html5-drag-drop-shim/pull/31/files#r23481352> .  <https://github.com/notifications/beacon/AAWWlfYc_b7X-nDGPornQcWyVMMQJuqOks5nkrNkgaJpZM4DWV2L.gif>